front_end/dart_tests/dart_support_test.dart

```diff
@@ -1,45 +1,170 @@
-/// Test the dart support functions standalone.
-
-// TODO(alanknight): How do we test things that only run in the context of
-// devtools?
-
-@JS()
 library dart_support_test;
 
 @TestOn('chrome')
 import 'package:test/test.dart';
 import 'package:js/js.dart';
 import 'dart:html';
 import 'dart:async';
+// TODO: Would it be better to pre-compile this and load it from a script tag?
+import 'context_for_evaluation.dart' as context;
+import 'js_interop_for_context_testing.dart';
+import 'test_helpers.dart';
 
-@JS()
-external String $dartExpressionFor(context, String expression);
-
-@JS('eval')
-external jsEval(String expression);
-
-unmodified(String s) {
-  var result = $dartExpressionFor(null, s);
+/// Assert that the dart expression for [s] is [s].
+unmodified(String s) async {
+  var result = await dartExpressionFor(null, s);
   expect(result.split('\n').last.trim(), s);
 }
 
-main() async {
+/// Create a script element for [path] to load it.
+ScriptElement loadScript(String path) {
+  // Note that we have to do this in a separate non-async method
+  // because of https://github.com/dart-lang/sdk/issues/35020
+  var element = new ScriptElement()
+    ..type = 'text/javascript'
+    ..async = false
+    ..src = path;
+  document.body.append(element);
+  return element;
+}
 
+/// Keep a global flag to see if we've loaded the JS scripts.
+bool loaded = false;
+
+main() async {
   setUp(() async {
-    var element = new ScriptElement()
-      ..type = 'text/javascript'
-      ..src = 'packages/dart_devtools/sdk/DartSupport.js';
-    document.body.append(element);
-    await new Future.delayed(new Duration(seconds: 2));
-    await element.onLoad.take(1);
+    // Load the Devtools Dart code and our support script.
+    if (loaded) return new Future.value(null);
+    var scripts = [
+      'sdk/DartSupport.js',
+      'sdk/DartEval.js',
+      'sdk/DartScopes.js',
+      'test_support.js'
+    ].map((x) => 'packages/dart_devtools/$x');
+    for (var s in scripts) {
+      await loadScript(s).onLoad.first;
+    }
+    loaded = true;
+  });
+
+  test("Not Dart code bails out", () {
+    unmodified('(a/&*');
   });
 
-  test("Doesn't look like Dart", () {
-    unmodified('a/&*');
+  test("Basic evaluating in a context", () {
+    // The 'context' library has hard-coded evaluation points. Make sure that we
+    // can evaluate a simple expression.
+    jsSetThingToEval('2+2');
```

Comment:
nit: make your basic expression something that wouldn't work with JS eval.
e.g. test that
'a' * 3
returns 'aaa'